### ISSUE TITLE
Incorrect handling of proxy imports

### ISSUE THREAD

**Issue by AMDmi3** [python/mypy#7067]
Regression in 0.710 (still not fixed in 0.711). Not sure how it's called properly, but the problem is that mypy doesn't properly handle symbols imported from a package which imports them too. Example:

provider.py
```python
def foobar1() -> None:
    pass
def foobar2() -> None:
    pass
def foobar3() -> None:
    pass
```
proxy.py
```python
from provider import foobar1, foobar2, foobar3
```
main.py
```python
from proxy import foobar1, foobar2, foobar3
```

`main.py` runs without errors and mypy 0.701 doesn't complain. However mypy 0.710 does:
```
main.py:1: error: Module 'proxy' has no attribute 'foobar1'; maybe "foobar3" or "foobar2"?
main.py:1: error: Module 'proxy' has no attribute 'foobar2'; maybe "foobar3" or "foobar1"?
main.py:1: error: Module 'proxy' has no attribute 'foobar3'; maybe "foobar2" or "foobar1"?
```
It's funny it complains on the very names it suggests. Adding `__all__` to `proxy.py` doesn't help (however IMO it should work regardless).

**Comment by JelleZijlstra** [comment]
Are you using `--strict`? This sounds like the behavior of the `--no-implicit-reexport ` flag (documented at https://mypy.readthedocs.io/en/latest/command_line.html#miscellaneous-strictness-flags), which is included in `--strict`.

**Comment by AMDmi3** [comment]
Yes, I am. Thanks for the hint and sorry for the noise. I'm seeing this problem in third party code too, will link them to this issue.
